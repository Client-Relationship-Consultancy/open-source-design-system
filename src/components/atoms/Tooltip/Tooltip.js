import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import ReactTooltip from "react-tooltip"
import TooltipIcon from "assets/tooltip-icon.svg"

const Container = styled.div`
  display: inline-flex;
  flex-direction: row;
  flex-basis: content;
  align-items: center;
`

const Icon = styled.img`
  width: 18px;
  height: 18px;
  margin-left: 5px;
`

const ToolTipContainer = styled.div`
  > div.visible-on-hover {
    pointer-events: auto !important;
    &:hover {
      visibility: visible !important;
    }
  }
`

const Tooltip = props => (
  <>
    <Container data-tip data-for={props.id}>
      {props.children}
      {props.showIcon && <Icon src={TooltipIcon} />}
    </Container>
    <ToolTipContainer className={props.className}>
      <ReactTooltip
        className="visible-on-hover"
        delayHide={500}
        effect="solid"
        type="success"
        id={props.id}
      >
        {props.tooltipContent}
      </ReactTooltip>
    </ToolTipContainer>
  </>
)

Tooltip.defaultProps = {
  showIcon: true,
  tooltipContent: () => null,
}

Tooltip.propTypes = {
  children: PropTypes.node,
  tooltipContent: PropTypes.node,
  showIcon: PropTypes.bool,
  id: PropTypes.string,
  className: PropTypes.string,
}

export default Tooltip
