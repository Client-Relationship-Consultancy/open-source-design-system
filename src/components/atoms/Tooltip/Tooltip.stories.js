import React from "react"
import { storiesOf } from "@storybook/react"

import { Tooltip } from "../../../index"

const stories = storiesOf("Tooltip", module)

const ExampleContent = () => <span>I am inside a tooltip, look at me!!!!</span>

stories.add("Tooltip", () => <Tooltip tooltipContent={<ExampleContent />}>Hover over me</Tooltip>)
