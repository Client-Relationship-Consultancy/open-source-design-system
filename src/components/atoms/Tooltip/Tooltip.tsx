import React from "react";
import styled from "styled-components";
import ReactTooltip from "react-tooltip";
import TooltipIcon from "../../../assets/tooltip-icon.svg";

interface IProps {
  showIcon?: boolean;
  id: string;
  className?: string;
  tooltipContent?: React.ReactNode;
}
const Container = styled.div`
  display: inline-flex;
  flex-direction: row;
  flex-basis: content;
  align-items: center;
`;

const Icon = styled.img`
  width: 18px;
  height: 18px;
  margin-left: 5px;
`;

const ToolTipContainer = styled.div`
  > div.visible-on-hover {
    pointer-events: auto !important;
    &:hover {
      visibility: visible !important;
    }
  }
`;

const Tooltip: React.FunctionComponent<IProps> = props => (
  <>
    <Container data-tip data-for={props.id}>
      {props.children}
      {props.showIcon && <Icon src={TooltipIcon} />}
    </Container>
    <ToolTipContainer className={props.className}>
      <ReactTooltip
        className="visible-on-hover"
        delayHide={500}
        effect="solid"
        type="success"
        id={props.id}
      >
        {props.tooltipContent || null}
      </ReactTooltip>
    </ToolTipContainer>
  </>
);

Tooltip.defaultProps = {
  showIcon: true,
};

export default Tooltip;
