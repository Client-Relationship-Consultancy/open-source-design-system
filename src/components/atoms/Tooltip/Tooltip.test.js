import React from "react"
import { shallow } from "enzyme"

import Tooltip from "./Tooltip"

const ExampleContent = () => <span>I am inside a tooltip, look at me!!!!</span>

describe("Tooltip Component Testing", () => {
  const component = shallow(<Tooltip tooltipContent={<ExampleContent />}>Hover over me</Tooltip>)
  it("should find icon", () => {
    expect(component.find("Tooltip__Icon").exists())
  })
  it("should not show icon if showIcon prop is false", () => {
    const noIconComponent = shallow(
      <Tooltip tooltipContent={<ExampleContent />} showIcon={false}>
        There is no icon here
      </Tooltip>,
    )
    expect(noIconComponent.find("Tooltip__Icon").exists()).toBe(false)
  })
  it("should match latest snapshot", () => {
    expect(component).toMatchSnapshot()
  })
})
