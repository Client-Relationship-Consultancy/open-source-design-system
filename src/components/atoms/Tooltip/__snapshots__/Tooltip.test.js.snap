// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Component Testing should match latest snapshot 1`] = `
<Fragment>
  <Tooltip__Container
    data-tip={true}
  >
    Hover over me
    <Tooltip__Icon
      src={null}
    />
  </Tooltip__Container>
  <Tooltip__ToolTipContainer>
    <ReactTooltip
      className="visible-on-hover"
      clickable={false}
      delayHide={500}
      effect="solid"
      insecure={true}
      resizeHide={true}
      type="success"
      wrapper="div"
    >
      <ExampleContent />
    </ReactTooltip>
  </Tooltip__ToolTipContainer>
</Fragment>
`;
